### PR TITLE
New random mission

### PR DESCRIPTION
**Pull request by El-Jekozo**
#### Summary

SUMMARY: Content "New random mission: Angry pyromaniac"

#### Purpose of change

This PR adds a new random mission for NPCs: "Angry pyromaniac".

#### Describe the solution

Added a new random mission for random NPCs. NPC will ask player to burn tagged house. Then he will start following player after accepting mission. At the end of the mission, fire will be placed on tagged tile (to simulate burning) and NPC will join player faction.

#### Describe alternatives you've considered

None.

#### Testing

I've tested it in game, works as intended. One little problem. Because of how MGOAL_GO_TO works (it counts as succeeded by standing on tagged tile OR on tile near it), it's possible to complete mission and "burn" tile near tagged.

#### Additional context

None.

### CODE REVIEW

File: data/json/npcs/TALK_COMMON_MISSION.json [hunk 1]
```diff
@@ -324,5 +324,19 @@
       { "text": "Thank you.", "topic": "TALK_NONE", "effect": "clear_mission" },
       { "text": "Thanks, bye.", "topic": "TALK_DONE", "effect": "clear_mission" }
     ]
+  },
+  {
+    "//": "MISSION_PYROMANIAC mission set this topic as starting topic for mission giver.",
+    "type": "talk_topic",
+    "id": "MISSION_PYROMANIAC",
+    "dynamic_line": "Are we there yet? I can't wait to burn that house!",
```
Comment (ZhilkinSerg) [inline data/json/npcs/TALK_COMMON_MISSION.json:332]:
```suggestion
    "dynamic_line": "Are we there yet?  I can't wait to burn that house!",
```

File: data/json/npcs/TALK_COMMON_MISSION.json [hunk 2]
```diff
@@ -324,5 +324,19 @@
...
+    "responses": [
+      {
+        "text": "We're here. Let's do it!",
```
Comment (ZhilkinSerg) [inline data/json/npcs/TALK_COMMON_MISSION.json:335]:
```suggestion
        "text": "We're here.  Let's do it!",
```

File: data/json/npcs/TALK_COMMON_MISSION.json [hunk 3]
```diff
@@ -324,5 +324,19 @@
...
+        "topic": "TALK_MISSION_INQUIRE",
+        "condition": { "and": [ "mission_complete", { "u_has_item": "gasoline", "count": 200 } ] }
```
Comment (ZhilkinSerg) [inline data/json/npcs/TALK_COMMON_MISSION.json:337]:
```suggestion
        "condition": { "and": [ "mission_complete", { "u_has_items": "gasoline", "count": 200 } ] }
```

File: data/json/npcs/missiondef.json [hunk 4]
```diff
@@ -795,5 +795,53 @@
...
+        "z": 0
+      }
+    },
+    "end": {
+      "effect": [ "follow", { "u_consume_item": "gasoline", "count": 200 }, { "mapgen_update": "MISSION_PYROMANIAC_BURN" } ]
+    },
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "dialogue": {
+      "describe": "Oh man, I want to burn it so bad…",
+      "offer": "I'm so infuriated! I've got an enemy that ruined my life, and now I want to get revenge! I don't care about <the_cataclysm>, I just want to burn his house! Would you help me?",
```
Comment (ZhilkinSerg) [inline data/json/npcs/missiondef.json:827]:
```suggestion
      "offer": "I'm so infuriated!  I've got an enemy that ruined my life, and now I want to get revenge!  I don't care about <the_cataclysm>, I just want to burn his house!  Would you help me?",
```

File: data/json/npcs/missiondef.json [hunk 5]
```diff
@@ -795,5 +795,53 @@
...
+      "accepted": "Good. Let's go to his house and burn it down! Oh, by the way, could you bring gasoline with you? I was so angry I forgot to bring it with me…",
```
Comment (ZhilkinSerg) [inline data/json/npcs/missiondef.json:828]:
```suggestion
      "accepted": "Good.  Let's go to his house and burn it down!  Oh, by the way, could you bring gasoline with you?  I was so angry I forgot to bring it with me…",
```

File: data/json/npcs/missiondef.json [hunk 6]
```diff
@@ -795,5 +795,53 @@
...
+      "rejected": "Seriously? It's such an easy job…",
```
Comment (ZhilkinSerg) [inline data/json/npcs/missiondef.json:829]:
```suggestion
      "rejected": "Seriously?  It's such an easy job…",
```

File: data/json/npcs/missiondef.json [hunk 7]
```diff
@@ -795,5 +795,53 @@
...
+    },
+    "origins": [ "ORIGIN_OPENER_NPC" ],
+    "dialogue": {
+      "describe": "Oh man, I want to burn it so bad…",
+      "offer": "I'm so infuriated!  I've got an enemy that ruined my life, and now I want to get revenge!  I don't care about <the_cataclysm>, I just want to burn his house!  Would you help me?",
+      "accepted": "Good.  Let's go to his house and burn it down!  Oh, by the way, could you bring gasoline with you?  I was so angry I forgot to bring it with me…",
+      "rejected": "Seriously?  It's such an easy job…",
+      "advice": "Maybe we can find some gasoline at gas station.",
+      "inquire": "Are you ready?",
+      "success": "Sweet, sweet revenge! Ah, smells so nice! Feels like the smell of napalm in the morning! All his stuff, and probably even himself, will burn to ashes in a matter of minutes.  You helped me get my revenge, so I'll follow you to the end, my friend!",
```
Comment (ZhilkinSerg) [inline data/json/npcs/missiondef.json:832]:
```suggestion
      "success": "Sweet, sweet revenge!  Ah, smells so nice!  Feels like the smell of napalm in the morning!  All his stuff, and probably even himself, will burn to ashes in a matter of minutes.  You helped me get my revenge, so I'll follow you to the end, my friend!",
```

File: data/json/npcs/TALK_COMMON_MISSION.json [hunk 8]
```diff
@@ -324,5 +324,19 @@
...
+  {
+    "//": "MISSION_PYROMANIAC mission set this topic as starting topic for mission giver.",
+    "type": "talk_topic",
+    "id": "MISSION_PYROMANIAC",
+    "dynamic_line": "Are we there yet, <name_g>?  I can't wait to burn that building!",
+    "responses": [
+      {
+        "text": "We're here.  Let's do it!",
+        "topic": "TALK_MISSION_INQUIRE",
+        "condition": { "and": [ "mission_complete", { "u_has_items": "gasoline", "count": 200 } ] }
```
Comment (ZhilkinSerg) [inline data/json/npcs/TALK_COMMON_MISSION.json:337]:
```suggestion
        "condition": { "and": [ "mission_complete", { "u_has_items":  { "item": "gasoline", "count": 200 } } ] }
```

**Review by Maleclypse — COMMENTED** [review]
So does the pyro set fire to the ground under the PC?  Doesn't that set the PC on fire?

**Comment by El-Jekozo** [comment]
> 
> 
> So does the pyro set fire to the ground under the PC? Doesn't that set the PC on fire?

It updates overmap tile by placing fire and spilled gasoline at random places of that tile. It MAY set PC or NPC on fire if PC or NPC would stand on map tile where fire would be placed.

**Review by Maleclypse — APPROVED** [review]

**Pull request merged**